python/iwp/data_loader.py

```diff
@@ -513,3 +515,64 @@ def add_timestep_as_coord( ds ):
                             preprocess=add_timestep_as_coord )
 
     return ds
+
+
+    def get_data_loader( dataset_path_pattern,
```

Comment:
Why not provide a class method that acts as a DataLoader factory?  You take a superset of arguments to initialize a Dataset and have three arguments (batch_size, shuffle_flag, number_workers) to create the DataLoader.

It may be poor design to create a DataLoader from a Dataset, though you're better positioned to understand the separation between those two and which approach is more natural.  I'd recommend looking at some PyTorch models (pick a model zoo) and see what patterns exist, if any, to help make this decision.